### PR TITLE
Show number of deleted events

### LINKED ISSUE #53: delete action - verify successful vs unsuccessful delete before returning OK
currently we always return that the delete was successfull but we need to verify this by checking if the entry exists or not.

### PR DESCRIPTION
**Pull request by mattiaslundberg**
Fix #53

### CODE REVIEW

File: chalicelib/action.py [hunk 1]
```diff
@@ -357,14 +357,31 @@ def perform_action(self):
...
-            self.send_attachment(
-                attachment=delete_message_menu(self.user_name, date_string)
+        if self._check_locks(date=date["from"], second_date=date["to"]):
+            return self.send_response(
+                message=f"Unable to delete since month is locked :cry:"
             )
-        else:
-            self.send_response(message=f"Unable to delete since month is locked :cry:")
+
+        if not self._has_events(date["from"]):
```
Comment (parberge) [inline chalicelib/action.py:365]: We have `_get_events()` method that we might use instead.

And then you can remove the `_has_events` method, or simplify it by reusing `_get_events()` in that method

File: chalicelib/action.py [hunk 2]
```diff
@@ -357,14 +362,29 @@ def perform_action(self):
...
 
-        if not self._check_locks(date=date["from"], second_date=date["to"]):
-            self.send_attachment(
-                attachment=delete_message_menu(self.user_name, date_string)
+        if self._check_locks(date=date["from"], second_date=date["to"]):
+            return self.send_response(
+                message=f"Unable to delete since month is locked :cry:"
+            )
+
+        if not self._has_events(date["from"]):
```
Comment (parberge) [inline chalicelib/action.py:370]: Use the existing date_string instead
```suggestion
        if not self._has_events(date=date_string):
```

The other suggestions is depending on this change

File: chalicelib/action.py [hunk 3]
```diff
@@ -357,14 +362,29 @@ def perform_action(self):
...
-        else:
-            self.send_response(message=f"Unable to delete since month is locked :cry:")
+
+        self.send_attachment(
+            attachment=delete_message_menu(self.user_name, date_string)
+        )
         return ""
 
+    def _has_events(self, date: datetime):
+        date_str = date.strftime(self.format_str)
```
Comment (parberge) [inline chalicelib/action.py:381]:
```suggestion
    def _has_events(self, date: str):
```

File: chalicelib/action.py [hunk 4]
```diff
@@ -357,14 +362,29 @@ def perform_action(self):
...
+        events = self._get_events(date_str=date_str)
```
Comment (parberge) [inline chalicelib/action.py:382]:
```suggestion
        events = self._get_events(date_str=date)
```

File: chalicelib/action.py (hunk 2 above)
Comment (mattiaslundberg) [reply chalicelib/action.py:370]: Doesn't this break `delete today`? `date_string` is the data as written in slack and it seems the api doesn't support `today`. Even if we're okay with removing `today` I prefer to use the cleaned data before passing it (harder to pass invalid data trough)

Comment (parberge) [reply chalicelib/action.py]: Good point

**Pull request merged**